### ISSUE TITLE
Incorrect summary displayed on map page for some fields (at least maturity)

### ISSUE THREAD

**Issue by omad** [opendatacube/datacube-explorer#431]
Within DEA we have just switched to generating `interim` quality ARD data instead of `final` quality for Landsat 8, due to the recent unavailability of BRDF.

When I go looking for this information, Explorer is presenting an inconsistent view.

On the dataset listing page, I correctly see a list of both `final` and `interim` datasets:
![image](https://user-images.githubusercontent.com/108979/186341279-5ae79207-1beb-424e-8275-fd9caf9deb77.png)

However, when I view the same date on the Map/Summary page, it indicates that all datasets are of `final` quality.
![image](https://user-images.githubusercontent.com/108979/186341331-c18dc0e6-7689-43c6-9881-a83e200dcede.png)

**Comment by jeremyh** [comment]
Yeah, it is a little misleading. The little squiggly underline on the word is because Explorer is guessing -- it's based on a sampling of the datasets. So all datasets it sampled consistently had that property.

There's a little tooltip:
![Screenshot from 2022-08-25 09-49-40](https://user-images.githubusercontent.com/25688/186543750-e8e5c3f0-1529-4098-92f8-5d5b5dbaaf79.png)

The non-squiggly-underlined ones are actually concrete rules on the product

**Comment by pindge** [comment]
## Proposed solution
1. add a description to the block, indicating the sample size
2. instead of comparing with the first dataset, search for the latest indexed datasets https://github.com/opendatacube/datacube-explorer/blob/a00bcb04884d4350f2a21f9a4cfa96ed8d008c09/cubedash/summary/_stores.py#L631
